Modules/Devices/src/Devices.Domain/Entities/Identity.cs

```diff
@@ -25,6 +27,8 @@ public Identity(string? clientId, IdentityAddress address, byte[] publicKey, byt
 
     public byte IdentityVersion { get; set; }
 
+    public Tier? Tier { get; set; }
```

Comment:
@tnotheis We have a chicken and egg problem here. The database may already have Identities, which do not have a Tier. When we add a TierId to the Identities table (via Migrations), we must set that collumn as nullable, hence that code. 
The alternative - having a default value - is inviable because the Basic Tier may not exist yet.
How should we address this?

Reply:
A possible solution seems to be to make the attribute mandatory but tweak the migration to make the column nullable. I do not know how that can impact the runtime of the application though.

Reply:
Okay, let's try to go with your suggested approach: make the property mandatory and make the column nullable. But don't edit the migration directly. Instead, you should edit the `IdentityEntityTypeConfiguration`:

```c#
builder.Property(x=>x.Tier).IsRequired(false)
```
I'm not sure though if that works in case of a navigation property. If not, I would suggest to leave it as it is for now, and try to remember to fix it in the future when the possibility that there are still people running a version without Tiers is almost 0.

Reply:
We're using `.HasOne()`, which does not support `IsRequired` nor anything similar.

This SO answer seems to shed some light on the issue at hand: https://stackoverflow.com/a/35562728

Reply:
Yeah, okay. Let's just ignore that for now. It's too much pain for too less benefit.

Reply:
No worries. I would have done it myself. 😆